### DISCUSSION TITLE
Can't get Namecheap IP to update for domain

### DISCUSSION THREAD

**Discussion by frequentbeef** [qdm12/ddns-updater#454] [Q&A] [answered]
I've gotten things to the point where the log for the container say that the IP record has been updated and web interface also reports that it was updated (and saves the action in the updates.json file), but it never actually seems to get updated in Namecheap.

The IP listed with the DNS record never changes, and thus the container keeps thinking that it's updated it every 5 minutes.

I think I followed the process properly in both configuring the container and Namecheap, so I'm at a loss on what to do next or how to troubleshoot it.

I am running this on a Synology NAS, which presented some challenges in getting the right folder permissions, but it's reading/writing to it now, so that seems resolved.

**Comment by frequentbeef** [comment]
I tried configuring a Cloudflare setup for another domain and can't get that one to work, either, but I get an error, at least?

When I created an API token specifically with DNS edit permissions for the zone of the domain and ran the container, the log reports and error of: ERROR cannot get record ID: no result received

When I swapped that token out for my global API key (which was referenced elsewhere), I got a different error: ERROR[0m cannot get record ID: bad HTTP status: 400: {"success":false,"errors":[{"code":10000,"message":"Authentication error"}]}

**Comment by frequentbeef** [comment, marked as answer]
I kept poking at it and managed to get it working - I had to manually create a record in Cloudflare to be updated and then I realized my mistake with Namecheap was I had mistyped the domain.

**Reply by qdm12** [reply]
Thanks for answering 😉
I'll check Cloudflare if the record can be created if it doesn't exist, that would be better.

**Reply by frequentbeef** [reply]
Yeah, I figured I needed to at least come back and clean up my mess in case someone else runs into the same issues. Namecheap was me just making a basic mistake but Cloudflare was more me not knowing what to look for or do, since I couldn't find as good of a step-by-step guide. I had used a DynDNS.org updater before and I think it just took care of a lot of that for me.

Even if it can't create the record if it's not there, the error message pointing me to the Cloudflare site to set up a record instead of trying to find a record ID or wonder why it wasn't saving a record locally, etc. would have helped.

I'm sure it would have been more obvious to folks with more experience :)

**Reply by qdm12** [reply]
Cloudflare records are now created if they don't exist 😉
